### PR TITLE
docs(u-boot): AM335X: Update splash screen details for 09.03.05.02

### PR DESCRIPTION
**Pull request by sukrutb**
Update splash screen details with defconfig changes to enable splash screen.
Add steps to test the splash screen.
Add the reference output image on AM335x EVM.

### CODE REVIEW

**Comment by praneethbajjuri** [comment]
@devarsht , @s-jain1 please review

File: source/linux/Foundational_Components/U-Boot/Apps-Splash-Screen.rst [hunk 1]
```diff
@@ -10,247 +10,62 @@ U-Boot Splash Screen
...
-    #endif
+   #To load image
+   => fatload mmc 0 $loadaddr ti_logo_414x97_32bpp.bmp.gz
 
-In arch/arm/cpu/armv7/am33xx/clock\_am33xx.c enable the LCDC clocks.
+   #To display image
+   =>  bmp display $loadaddr m m
 
-::
+The above command will display an image at the centre of the screen as below
```
Comment (StaticRocket) [inline source/linux/Foundational_Components/U-Boot/Apps-Splash-Screen.rst:60]:
```suggestion
The above command will display an image at the center of the screen as shown below
```

Style: center is used substantially more than centre

File: source/linux/Foundational_Components/U-Boot/Apps-Splash-Screen.rst [hunk 2]
```diff
@@ -10,247 +10,62 @@ U-Boot Splash Screen
...
-                    &cmper->emifclkctrl,
-                    &cmper->lcdclkctrl,
-                    &cmper->lcdcclkstctrl,
-                    &cmper->epwmss2clkctrl,
-                    0
+.. Image:: /images/AM335x_evm_uboot_splash_screen.jpg
+   :width: 600
 
-In board.c add includes for mmc, fat, lcd, and the frame buffer.
+User can use the custom image to display the splash screen.
```
Comment (StaticRocket) [inline source/linux/Foundational_Components/U-Boot/Apps-Splash-Screen.rst:65]:
```suggestion
The user may use any bitmap image as splash screen. The below command may be used to verify the image is valid.
```

Even this is a little vague. EFI boot splash on amd64 still has general requirements about image size, alpha channel usage, etc. There needs to be more information here.

File: source/linux/Foundational_Components/U-Boot/Apps-Splash-Screen.rst [hunk 3]
```diff
@@ -10,247 +10,62 @@ U-Boot Splash Screen
 .. rubric:: AM335x
    :name: am335x
 
-All the code below is based on Processor Linux SDK 03.02.00..05.
+A splash screen is an introductory screen the user sees over the display when the device boots up. Splash screens are typically used to provide feedback to the user that the board is booting up and to showcase
+the vendor logo which helps with branding and identification of which boot image is being used for the bootup.
+All the code related to u-boto splash screen support is based on Processor Linux SDK 09.03.05.02.
```
Comment (StaticRocket) [inline source/linux/Foundational_Components/U-Boot/Apps-Splash-Screen.rst:15]:
```suggestion
All the code related to u-boot splash screen support is based on Processor Linux SDK 09.03.05.02.
```

File: source/linux/Foundational_Components/U-Boot/Apps-Splash-Screen.rst [hunk 4]
```diff
@@ -10,247 +10,62 @@ U-Boot Splash Screen
...
 
-There is a frame buffer driver for am335x in the drivers/video directory
-called am3355x-fb.c. It makes calls to routines in board.c to set up the
-LCDC and frame buffer. To use it:
+For AM335x, splash screen is enabled in u-boot which uses the LCDC driver.
+On AM335x evm, the LCD backlight is controlled using enhanced Auxiliary(eCAP0) PWM.
+To use this, enable following configurations in configs/am335x_evm_defconfig.
```
Comment (StaticRocket) [inline source/linux/Foundational_Components/U-Boot/Apps-Splash-Screen.rst:19]:
```suggestion
To use this, enable following configurations in :file:`configs/am335x_evm_defconfig`.
```

**Review by StaticRocket — CHANGES_REQUESTED** [review]

File: source/linux/Foundational_Components/U-Boot/Apps-Splash-Screen.rst [hunk 5]
```diff
@@ -10,247 +10,62 @@ U-Boot Splash Screen
...
 
-Either create a new defconfig in the configs directory or just add
-SPLASH to CONFIG\_SYS\_EXTRA\_OPTIONS. In this example the
-am335x\_evm\_defconfig is copied into a new one called
-am335x\_evm\_splash\_defconfig.
+.. code-block:: kconfig
 
-.. code-block:: text
+    CONFIG_SPLASH_SCREEN=y
```
Comment (cshilwant) [inline source/linux/Foundational_Components/U-Boot/Apps-Splash-Screen.rst:23]: The indentation of code sections should clearly line up with code-block directive as mentioned in CONTRIBUTING guidelines - https://github.com/TexasInstruments/processor-sdk-doc/blob/master/CONTRIBUTING.md#indentation-and-whitespace

File: source/linux/Foundational_Components/U-Boot/Apps-Splash-Screen.rst [hunk 6]
```diff
@@ -10,247 +10,62 @@ U-Boot Splash Screen
...
+.. code-block:: console
 
-    /* Splash scrren support */
-    #ifdef CONFIG_SPLASH
-    #define CONFIG_AM335X_LCD
-    #define CONFIG_LCD
-    #define CONFIG_LCD_NOSTDOUT
-    #define CONFIG_SYS_WHITE_ON_BLACK
-    #define LCD_BPP LCD_COLOR16
+   #To see all the files in your boot partition run
```
Comment (cshilwant) [inline source/linux/Foundational_Components/U-Boot/Apps-Splash-Screen.rst:51]: Same comment of indentation of code sections as above

File: source/linux/Foundational_Components/U-Boot/Apps-Splash-Screen.rst
Comment (cshilwant) [inline source/linux/Foundational_Components/U-Boot/Apps-Splash-Screen.rst:1]: If you can, please remove processors wiki link from line 1 - https://github.com/TexasInstruments/processor-sdk-doc/pull/116/files#diff-a5d6eca7afe4291aa3788e3750f15a0d7a4452bfdfb1b14aa98f64c5dbe43dc5R1

**Review by cshilwant — CHANGES_REQUESTED** [review]

**Comment by praneethbajjuri** [comment]
@sukrutb  please update

**Review by cshilwant — CHANGES_REQUESTED** [review]
Please plan to use ref and cross reference this section under AM335x release notes.

Checkout following links for more details,

How to use ref - https://www.sphinx-doc.org/en/master/usage/referencing.html#cross-referencing-arbitrary-locations

Link to AM335X release notes - https://github.com/TexasInstruments/processor-sdk-doc/blob/master/source/devices/AM335X/linux/Release_Specific_Release_Notes.rst?plain=1#L60

**Review by devarsht — COMMENTED** [review]
@sukrutb @praneethbajjuri 

Ain't it more apt to use UG-Splash screen i.e. below instead of APP-Splash-screen, For all Sitara platforms we use UG-Splash screen so that guide is more evolved.
 https://github.com/TexasInstruments/processor-sdk-doc/blob/0cbc0f59f3aea10940952042012043c21c232fc7/source/linux/Foundational_Components/U-Boot/UG-Splash-Screen.rst

File: source/linux/Foundational_Components/U-Boot/Apps-Splash-Screen.rst (hunk 5 above)
Comment (sukrutb) [reply source/linux/Foundational_Components/U-Boot/Apps-Splash-Screen.rst:23]: I will fix this.

File: source/linux/Foundational_Components/U-Boot/Apps-Splash-Screen.rst (hunk 6 above)
Comment (sukrutb) [reply source/linux/Foundational_Components/U-Boot/Apps-Splash-Screen.rst:51]: sure.

**Comment by sukrutb** [comment]
> @sukrutb @praneethbajjuri
> 
> Ain't it more apt to use UG-Splash screen i.e. below instead of APP-Splash-screen, For all Sitara platforms we use UG-Splash screen so that guide is more evolved. https://github.com/TexasInstruments/processor-sdk-doc/blob/0cbc0f59f3aea10940952042012043c21c232fc7/source/linux/Foundational_Components/U-Boot/UG-Splash-Screen.rst
@devarsht I am not familiar with the cross platform reference of the release notes. Can you please elaborate?
AFAIK, this is related to AM62x (specifically HW, u-boot SW, and supported features), which is not the same as AM33xx implementation(e.g., we don't have SPL splash screen support).
So, I have updated in the APP-Splash-screen.

File: source/linux/Foundational_Components/U-Boot/Apps-Splash-Screen.rst (hunk 2 above)
Comment (cshilwant) [reply source/linux/Foundational_Components/U-Boot/Apps-Splash-Screen.rst:65]: @sukrutb any comments on this?

**Review by cshilwant — DISMISSED** [review]
@sukrutb I still don't see this as well as other references of code section indentation under code-blocks being fixed - https://github.com/TexasInstruments/processor-sdk-doc/pull/116/files#r1887764585

**Comment by devarsht** [comment]
> > @sukrutb @praneethbajjuri
> > Ain't it more apt to use UG-Splash screen i.e. below instead of APP-Splash-screen, For all Sitara platforms we use UG-Splash screen so that guide is more evolved. https://github.com/TexasInstruments/processor-sdk-doc/blob/0cbc0f59f3aea10940952042012043c21c232fc7/source/linux/Foundational_Components/U-Boot/UG-Splash-Screen.rst
> > @devarsht I am not familiar with the cross platform reference of the release notes. Can you please elaborate?
> > AFAIK, this is related to AM62x (specifically HW, u-boot SW, and supported features), which is not the same as AM33xx implementation(e.g., we don't have SPL splash screen support).
> > So, I have updated in the APP-Splash-screen.

The UG-Splash-screen.rst is not specific to AM62x, other SoCs like AM62p too use the same guide. I am not sure about the goal for app-splash-screen.rst need to check with Bin on this.

**Review by devarsht — COMMENTED** [review]
@sukrutb , anything to add from perspective of flicker-free transition solution ? I remember you did some enablement using pwm preservation and some discussions were there with simple-framebuffer too ?

**Comment by cshilwant** [comment]
@sukrutb I see many unrelated commits. Can you rebase and ensure only intended commits are in your PR

**Comment by sukrutb** [comment]
> @sukrutb I see many unrelated commits. Can you rebase and ensure only intended commits are in your PR

@cshilwant , Done.

**Review by cshilwant — CHANGES_REQUESTED** [review]
@sukrutb I still see 5 commits on this PR.

I believe you used the Github's apply suggestion feature? 

Apparently that's not encouraged. Author's are expected to resolve comments from reviewers manually.

The Github's apply suggestion feature messes with the Signed-off-by tags

FYR, I see below in 12866399d8bd595cd6fdae2842b7319161aabae3

```sh
Signed-off-by: Sukrut Bellary <63480227+sukrutb@users.noreply.github.com>
```

**Comment by sukrutb** [comment]
> @sukrutb I still see 5 commits on this PR.
> 
> I believe you used the Github's apply suggestion feature?
> 
> Apparently that's not encouraged. Author's are expected to resolve comments from reviewers manually.
> 
> The Github's apply suggestion feature messes with the Signed-off-by tags
> 
> FYR, I see below in [1286639](https://github.com/TexasInstruments/processor-sdk-doc/commit/12866399d8bd595cd6fdae2842b7319161aabae3)
> 
> ```shell
> Signed-off-by: Sukrut Bellary <63480227+sukrutb@users.noreply.github.com>
> ```

@cshilwant , fixed. Please review. Thanks.

**Comment by cshilwant** [comment]
@praneethbajjuri @StaticRocket @devarsht for re-review

**Review by cshilwant — APPROVED** [review]

**Review by praneethbajjuri — APPROVED** [review]

**Pull request merged**